### PR TITLE
Bugfix - Editions

### LINKED ISSUE #480: Bug: Event Organizer usability when it comes to creating edition
> If editions is the important starting point to set right than have it as the first menu item. For user friendly might be a good thing to have the moderator menu items in the right order of hierarchy

> Edition: If something still needs to be done than maybe give it a color so it shows somethings missing.

> User friendly would be nice if I edit the events within an edition and changed one date it goes back to the edit page instead of the read page. Avoiding a lot of clicking back and forth

### PR DESCRIPTION
**Pull request by IGORnvk**
# Description

Reworked editions logic based on the feedback from the demo. Just test and see.

closes #480 #490 #504 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## What needs to be tested
After migrating and seeding, as an event organizer:

- [x] Try acessing any other moderator route, such as: `/schedule` and verify that it redirects to `moderator/editions` if there is no active edition.
- [x] Try the same with other roles (e.g. participant can't access any mod routes). Basically double check if all the routes are protected.
- [x] Verify that there is a badge next to `Editions` in sidemenu.
- [x] Click on the edition.
- [x] Set dates of both edition and events and verify that you can't make an edition "during the year Jesus was born". Try to break date inputs as much as possible.
- [x] Activate the edition.
- [x] Add/Edit keynote speaker on the show page of the edition.
- [x] Verify that you can access other moderator routes once the edition is activated.
- [x] Create new edition.
- [x] Delete edition.

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Middleware/EditionMiddleware.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        if (!Auth::user()->is_crew) {
+            abort(403);
+        }
+
+        if (!Edition::current()) {
+            return redirect(route('moderator.editions.index'))
+                ->banner("There is no active edition. You can't access this page for now.");
+        }
```
Comment (v-stamenova) [inline app/Http/Middleware/EditionMiddleware.php:31]: Use different banner:
```php
return redirect()->route('subscriptions')->warningBanner('Subscription pending approval.');
return redirect()->route('subscriptions')->dangerBanner('Subscription cancellation failed.');
```

File: routes/web.php
Comment (v-stamenova) [inline routes/web.php:1]: The '/register/company' and '/register/participant' are accessible even when an edition is not active

**Review by v-stamenova — DISMISSED** [review]
A bit tired and just looked through it real quick but I think those are just the main stuff

**Comment by TimKardol** [comment]
image
Frontend wise: I think it could be clearer that these are clickable

**Review by TimKardol — APPROVED** [review]

**Pull request merged**